### PR TITLE
Improve TagsPopup

### PR DESCRIPTION
**Pull request by gsdstr**
# Contribution checklist

## Brief description

Quick Add action Selecting a created item On enter action Fill label title from search input

Improvements are suitable as a reference for #4858

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does a local formatting is applied (rush format)
* [ ] - Does a local svelte-check is applied (rush svelte-check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

#4858

## Contributor requirements

* [x] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [x] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

<sub>View in Huly <a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NWU2ZjI3Yzk1ZmRhMGNjMDg5OWZhZWIiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.kjObU5OGBLxWlKcXM8-PD1NlEqFd88TI5XYi81hxO3g">UBERF-5883</a></sub>

### CODE REVIEW

File: plugins/tags-resources/src/components/CreateTagElement.svelte [hunk 1]
```diff
@@ -52,8 +54,13 @@
     color = getColorNumberByText(title)
   }
 
-  $: if (!categoryWasSet && categories.length > 0) {
-    category = findTagCategory(title, categories)
+  $: {
+    if (!categoryWasSet && categories.length > 0) {
+      category = findTagCategory(title, categories)
+      if (quick) { // TODO dirty solution, move all logic to separate class
```
Comment (aonnikov) [inline plugins/tags-resources/src/components/CreateTagElement.svelte]: Yeah, this is dirty. In fact, we don't need to use this component for "quick" label creation, it is enough to have an utility function for that.

File: plugins/tags-resources/src/components/CreateTagElement.svelte [hunk 2]
```diff
@@ -35,8 +35,10 @@
 
   export let targetClass: Ref<Class<Doc>>
   export let keyTitle: string = ''
+  export let prefillTitle: string = ''
```
Comment (aonnikov) [inline plugins/tags-resources/src/components/CreateTagElement.svelte]: not the best name for this variable. Can we use `keyTitle` or `title` instead?

File: plugins/tags-resources/src/components/TagsPopup.svelte [hunk 3]
```diff
@@ -101,6 +115,16 @@
     }
     return r
   }
+
+  async function onSearchKeydown (ev: KeyboardEvent): Promise<void> {
+    if (ev.code !== 'Enter') return
+    if (objects.length < 1) {
```
Comment (aonnikov) [inline plugins/tags-resources/src/components/TagsPopup.svelte]: minor: there is a chance that by the time user hits Enter, the query did not return new objects and `objects.length` is 0, but the the label exists. As a result, we create duplicate label.

**Review by aonnikov — COMMENTED** [review]
Thanks for your contribution! The idea is great, but need to do some additional changes.
And I'm not completely sure that we need "quick" label creation. In this approach there is no way for user to specify color for the label. Maybe instead of creating a label right away we need to show the user label creation dialog with pre populated label title. If he is ok with other label properties, he hit Enter once again and create the label. Think about it.

**Comment by gsdstr** [comment]
> And I'm not completely sure that we need "quick" label creation.

I think it's necessary to give the user the ability to edit created labels. This will speed up issues creation and keep the ability to choose colors. Like this:
Screenshot 2024-03-07 at 13 21 56

**Comment by aonnikov** [comment]
> > And I'm not completely sure that we need "quick" label creation.
> 
> I think it's necessary to give the user the ability to edit created labels. This will speed up issues creation and keep the ability to choose colors. Like this: Screenshot 2024-03-07 at 13 21 56

We definitely need to do this, but definitely not here. Tags/labels have larger scope than just tracker so there should be a more generic solution. Let's not focus on this right now. I propose to either go with "quick" creation or open tag creation dialog when user hits Enter.

**Review by aonnikov — DISMISSED** [review]

**Review by aonnikov — APPROVED** [review]

**Pull request merged**